### PR TITLE
TiUP: Add tiup-dm-topology-reference.md

### PR DESCRIPTION
**Pull request by qiancai**
### What is changed, added or deleted? (Required)

Add tiup-dm-topology-reference.md

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/4868
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiup/tiup-dm-topology-reference.md [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+
+# Topology Configuration File for DM Cluster Deployment using TiUP
+
+To deploy or scale a TiDB Data Migration (DM) cluster, you need to provide a topology file to describe the cluster topology. Similarly, to modify the cluster topology, you need to modify the topology file. The difference is that you can only modify a part of the fields in the topology file.
+
+Here is an [example topology file](https://github.com/pingcap/tiup/blob/master/embed/templates/examples/dm/topology.example.yaml) for your reference.
+
+## File Structure
+
+The topology file of a DM cluster might contain the following blocks:
```
Comment (ran-huang) [inline tiup/tiup-dm-topology-reference.md:13]: Refer to https://github.com/pingcap/docs/pull/5314/files#diff-33150822cf42e37369ebf2cd8dec35142aefda117ef5015af5755e6465d56d02R13
```suggestion
The topology file of a DM cluster might contain the following sections:
```

File: tiup/tiup-dm-topology-reference.md [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+
+- [global](/tiup/tiup-dm-topology-reference.md#global): the global configuration of the cluster. Some of the configuration items use the default values on the cluster, and you can configure them separately in each instance.
```
Comment (ran-huang) [inline tiup/tiup-dm-topology-reference.md:15]:
```suggestion
- [global](#global): the global configuration of the cluster. Some of the configuration items use the default values on the cluster, and you can configure them separately in each instance.
```
Self-referencing only needs an anchor name without a file name.

File: tiup/tiup-dm-topology-reference.md [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+- [alertmanager_servers](#alertmanager_servers): the configuration of the Alertemanager instances. The configuration specifies the machines to which the Alertmanager instances are deployed.
+
+### `global`
+
+The `global` section corresponds to the cluster's global configuration and has the following fields:
+
+- `user`: the user to start the deployed cluster. The default value is "tidb". If the user specified in the `<user>` field does not exist on the target machine, TiUP will automatically try to create the user.
+- `group`: the user group to which a user belongs when the user is automatically created. The default value is the same as the `<user>` field. If the specified group does not exist, it will be created automatically.
+- `ssh_port`: the SSH port to connect to the target machine for operations. The default value is "22".
+- `deploy_dir`: the deployment directory for each component. The default value is "deploy". The application rules are as follows:
```
Comment (AstroProfundis) [inline tiup/tiup-dm-topology-reference.md:30]:
```suggestion
- `deploy_dir`: the deployment directory for each component. The default value is "deploy". The construction rules are as follows:
```

File: tiup/tiup-dm-topology-reference.md [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+    - If the absolute path `deploy_dir` is configured at the instance level, the actual deployment directory is the `deploy_dir` configured for the instance.
+    - For each instance, if you do not configure `deploy_dir`, the default value is the relative path `<component-name>-<component-port>`.
+    - If `global.deploy_dir` is set to an absolute path, the component is deployed to `<global.deploy_dir>/<instance.deploy_dir>` directory.
+    - If `global.deploy_dir` is set to a relative path, the component is deployed to `/home/<global.user>/<global.deploy_dir>/<instance.deploy_dir>` directory.
+- `data_dir`: the data directory. The default value is "data". The application rules are as follows.
```
Comment (AstroProfundis) [inline tiup/tiup-dm-topology-reference.md:35]:
```suggestion
- `data_dir`: the data directory. The default value is "data". The construction rules are as follows.
```

File: tiup/tiup-dm-topology-reference.md [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+    - If the absolute path `data_dir` is configured at the instance level, the actual data directory is the `data_dir` configured for the instance.
+    - For each instance, if `data_dir` is not configured, the default value is `<global.data_dir>`.
+    - If `data_dir` is set to a relative path, the component data is stored in `<deploy_dir>/<data_dir>`. For the calculation rules of `<deploy_dir>`, see the application rules of the `deploy_dir` field.
```
Comment (AstroProfundis) [inline tiup/tiup-dm-topology-reference.md:38]:
```suggestion
    - If `data_dir` is set to a relative path, the component data is stored in `<deploy_dir>/<data_dir>`. For the construction rules of `<deploy_dir>`, see the construction rules of the `deploy_dir` field.
```

File: tiup/tiup-dm-topology-reference.md [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+- `log_dir`: the data directory. The default value is "log". The application rules are as follows.
```
Comment (AstroProfundis) [inline tiup/tiup-dm-topology-reference.md:39]:
```suggestion
- `log_dir`: the data directory. The default value is "log". The construction rules are as follows.
```

File: tiup/tiup-dm-topology-reference.md [hunk 7]
```diff
@@ -0,0 +1,293 @@
...
+    - If the absolute path of `log_dir` is configured at the instance level, the actual log directory is the `log_dir` configured for the instance.
+    - For each instance, if `log_dir` is not configured by the user, the default value is `<global.log_dir>`.
+    - If `log_dir` is a relative path, the component logs will be stored in `<deploy_dir>/<log_dir>`. For the calculation rules of `<deploy_dir>`, see the application rules of the `deploy_dir` field.
```
Comment (AstroProfundis) [inline tiup/tiup-dm-topology-reference.md:42]:
```suggestion
    - If `log_dir` is a relative path, the component logs will be stored in `<deploy_dir>/<log_dir>`. For the construction rules of `<deploy_dir>`, see the construction rules of the `deploy_dir` field.
```

File: tiup/tiup-dm-topology-reference.md [hunk 8]
```diff
@@ -0,0 +1,293 @@
...
+- `peer_port`: specifies the port for communication between DM masters. The default value is "8291".
+- `deploy_dir`: specifies the deployment directory. If the field is not specified, or specified as a relative directory, the deployment directory is generated according to the `deploy_dir` configuration in the `global` section.
+- `data_dir`: specifies the data directory. If the field is not specified, or specified as a relative directory, the data directory is generated according to the `data_dir` configuration in the `global` section.
+- `log_dir`: specifies the log directory. If the field is not specified, or specified as a relative directory, the log directory is generated according to the `log_dir` configuration in the `global` section.
+- `numa_node`: allocates the NUMA policy to the instance. Before specifying this field, you need to make sure that the target machine has [numactl](https://linux.die.net/man/8/numactl) installed. If this field is specified, cpubind and membind policies are allocated using [numactl](https://linux.die.net/man/8/numactl). This field is the string type. The field value is the ID of the NUMA node, such as "0,1".
```
Comment (AstroProfundis) [inline tiup/tiup-dm-topology-reference.md:95]:
```suggestion
- `numa_node`: allocates the NUMA policy to the instance. Before specifying this field, you need to make sure that the target machine has [numactl](https://linux.die.net/man/8/numactl) installed. If this field is specified, cpubind and membind policies are allocated using [numactl](https://linux.die.net/man/8/numactl). This field is a string type. The field value is the ID of the NUMA node, such as "0,1".
```

File: tiup/tiup-dm-topology-reference.md [hunk 9]
```diff
@@ -0,0 +1,293 @@
...
+- `port`: specifies the port on which DM worker provides services. The default value is "8262".
+- `deploy_dir`: specifies the deployment directory. If the field is not specified, or specified as a relative directory, the deployment directory is generated according to the `deploy_dir` configuration in the `global` section.
+- `data_dir`: specifies the data directory. If the field is not specified, or specified as a relative directory, the data directory is generated according to the `data_dir` configuration in the `global` section.
+- `log_dir`: specifies the log directory. If the field is not specified, or specified as a relative directory, the log directory is generated according to the `log_dir` configuration in the `global` section.
+- `numa_node`: allocates the NUMA policy to the instance. Before specifying this field, you need to make sure that the target machine has [numactl](https://linux.die.net/man/8/numactl) installed. If this field is specified, cpubind and membind policies are allocated using [numactl](https://linux.die.net/man/8/numactl). This field is the string type. The field value is the ID of the NUMA node, such as "0,1".
```
Comment (AstroProfundis) [inline tiup/tiup-dm-topology-reference.md:151]:
```suggestion
- `numa_node`: allocates the NUMA policy to the instance. Before specifying this field, you need to make sure that the target machine has [numactl](https://linux.die.net/man/8/numactl) installed. If this field is specified, cpubind and membind policies are allocated using [numactl](https://linux.die.net/man/8/numactl). This field is a string type. The field value is the ID of the NUMA node, such as "0,1".
```

File: tiup/tiup-dm-topology-reference.md [hunk 10]
```diff
@@ -0,0 +1,293 @@
...
+- `port`: specifies the port on which Prometheus provides services. The default value is "9090".
+- `deploy_dir`: specifies the deployment directory. If the field is not specified, or specified as a relative directory, the deployment directory is generated according to the `deploy_dir` configuration in the `global` section.
+- `data_dir`: specifies the data directory. If the field is not specified, or specified as a relative directory, the data directory is generated according to the `data_dir` configuration in the `global` section.
+- `log_dir`: specifies the log directory. If the field is not specified, or specified as a relative directory, the log directory is generated according to the `log_dir` configuration in the `global` section.
+- `numa_node`: allocates the NUMA policy to the instance. Before specifying this field, you need to make sure that the target machine has [numactl](https://linux.die.net/man/8/numactl) installed. If this field is specified, cpubind and membind policies are allocated using [numactl](https://linux.die.net/man/8/numactl). This field is the string type. The field value is the ID of the NUMA node, such as "0,1"
```
Comment (AstroProfundis) [inline tiup/tiup-dm-topology-reference.md:194]:
```suggestion
- `numa_node`: allocates the NUMA policy to the instance. Before specifying this field, you need to make sure that the target machine has [numactl](https://linux.die.net/man/8/numactl) installed. If this field is specified, cpubind and membind policies are allocated using [numactl](https://linux.die.net/man/8/numactl). This field is a string type. The field value is the ID of the NUMA node, such as "0,1"
```

File: tiup/tiup-dm-topology-reference.md [hunk 11]
```diff
@@ -0,0 +1,293 @@
...
+- `cluster_port`: Specify the port on which Alertmanger communicates with Alertmanagers. The default value is "9094".
+- `deploy_dir`: specifies the deployment directory. If the field is not specified, or specified as a relative directory, the deployment directory is generated according to the `deploy_dir` configuration in the `global` section.
+- `data_dir`: specifies the data directory. If the field is not specified, or specified as a relative directory, the data directory is generated according to the `data_dir` configuration in the `global` section.
+- `log_dir`: specifies the log directory. If the field is not specified, or specified as a relative directory, the log directory is generated according to the `log_dir` configuration in the `global` section.
+- `numa_node`: allocates the NUMA policy to the instance. Before specifying this field, you need to make sure that the target machine has [numactl](https://linux.die.net/man/8/numactl) installed. If this field is specified, cpubind and membind policies are allocated using [numactl](https://linux.die.net/man/8/numactl). This field is the string type. The field value is the ID of the NUMA node, such as "0,1"
```
Comment (AstroProfundis) [inline tiup/tiup-dm-topology-reference.md:268]:
```suggestion
- `numa_node`: allocates the NUMA policy to the instance. Before specifying this field, you need to make sure that the target machine has [numactl](https://linux.die.net/man/8/numactl) installed. If this field is specified, cpubind and membind policies are allocated using [numactl](https://linux.die.net/man/8/numactl). This field is a string type. The field value is the ID of the NUMA node, such as "0,1"
```

File: tiup/tiup-dm-topology-reference.md [hunk 12]
```diff
@@ -0,0 +1,293 @@
...
+- `numa_node`: allocates the NUMA policy to the instance. Before specifying this field, you need to make sure that the target machine has [numactl](https://linux.die.net/man/8/numactl) installed. If this field is specified, cpubind and membind policies are allocated using [numactl](https://linux.die.net/man/8/numactl). This field is the string type. The field value is the ID of the NUMA node, such as "0,1".
+- `config`: the configuration rules of this field are the same as that of `master` in the `server_configs` section. If `config` is specified, the configuration of `config` will be merged with the configuration of `master` in `server_configs` (if the two fields overlap, the configuration of this field takes effect), and then the configuration file is generated and distributed to the machine specified in the `host` field.
+- `os`: the operating system of the machine specified in the `host` field. If the field is not specified, the default value is the `os` value configured in the `global` section.
+- `arch`: the architecture of the machine specified by the `host` field. If the field is not specified, the default value is the `arch` value configured in the `global` section.
```
Comment (TomShawn) [inline tiup/tiup-dm-topology-reference.md:98]:
```suggestion
- `arch`: the architecture of the machine specified in the `host` field. If the field is not specified, the default value is the `arch` value configured in the `global` section.
```

**Review by TomShawn — COMMENTED** [review]
Good job! These are just minor edits.

File: tiup/tiup-dm-topology-reference.md [hunk 13]
```diff
@@ -0,0 +1,293 @@
...
+To deploy or scale a TiDB Data Migration (DM) cluster, you need to provide a topology file ([sample](https://github.com/pingcap/tiup/blob/master/embed/templates/examples/dm/topology.example.yaml)) to describe the cluster topology. 
+
+Similarly, to modify the cluster topology, you need to modify the topology file. The difference is that you can only modify a part of the fields in the topology file. This document introduces each section of the topology file and each field in each section.
+
+## File structure
+
+A topology configuration file for DM cluster deployment using TiUP might contain the following sections:
+
+- [global](#global): the cluster's global configuration. Some of the configuration items use the default values on the cluster, and you can configure them separately in each instance.
+- [server_configs](#server_configs): the components' global configuration. You can configure each component separately. If an instance has a configuration item with the same name, the instance's configuration item will take effect.
```
Comment (lucklove) [inline tiup/tiup-dm-topology-reference.md:16]:
```suggestion
- [server_configs](#server_configs): the components' global configuration. You can configure each component separately. If an instance has a configuration item with the same key, the instance's configuration item will take effect.
```

File: tiup/tiup-dm-topology-reference.md [hunk 14]
```diff
@@ -0,0 +1,293 @@
...
+
+A `monitoring_servers` configuration example is as follows:
+
+```yaml
+monitoring_servers:
+  - host: 10.0.1.11
+    rule_dir: /local/rule/dir
+```
+
+### ``grafana_servers``
```
Comment (ran-huang) [inline tiup/tiup-dm-topology-reference.md:219]:
```suggestion
### `grafana_servers`
```

File: tiup/tiup-dm-topology-reference.md [hunk 15]
```diff
@@ -0,0 +1,293 @@
...
+- `username`: specifies the username of the Grafana login screen.
+- `password`: specifies the corresponding password of Grafana.
+- `dashboard_dir`: specifies a local directory where the complete `dashboard(*.json)` files are located. The files in the specified directory will be sent to the target machine as Grafana dashboards during the initialization phase of the cluster configuration.
+- `resource_control`: resource control on this service. If this field is specified, the configuration of this field will be merged with the configuration of `resource_control` in the `global` section (if the two fields overlap, the configuration of this field takes effect), and then the configuration file of systemd is generated and distributed to the machine specified in the `host` field. The configuration rules of this field are the same as that of `resource_control` in the `global` section.
+
+> **Note:**
+>
+> If the `dashboard_dir` field of `grafana_servers` is configured, after executing the `tiup cluster rename` command to rename the cluster, you need to do the following:
```
Comment (ran-huang) [inline tiup/tiup-dm-topology-reference.md:236]:
```suggestion
> If the `dashboard_dir` field of `grafana_servers` is configured, after executing the `tiup cluster rename` command to rename the cluster, you need to perform the following operations:
```

File: tiup/tiup-dm-topology-reference.md [hunk 16]
```diff
@@ -0,0 +1,293 @@
...
+>
+> 1. In the local `dashboards` directory, update the value of the `datasource` field to the new cluster name (the `datasource` is named according to the cluster name).
```
Comment (ran-huang) [inline tiup/tiup-dm-topology-reference.md:238]:
```suggestion
> 1. In the local `dashboards` directory, update the value of the `datasource` field to the new cluster name (the `datasource` is named after the cluster name).
```

File: tiup/tiup-dm-topology-reference.md [hunk 17]
```diff
@@ -0,0 +1,293 @@
...
+    dashboard_dir: /local/dashboard/dir
+```
+
+### `alertmanager_servers`
+
+`alertmanager_servers` specifies the machines to which the Alertmanager service is deployed. You can also specify the service configuration on each machine. `alertmanager_servers` is an array. Each array element contains the following fields:
+
+- `host`: specifies the machine to deploy to. The field value is an IP address and is mandatory.
+- `ssh_port`: specifies the SSH port to connect to the target machine for operations. If the field is not specified, the `ssh_port` in the `global` section is used.
+- `web_port`: specify the port on which Alertmanager provides webpage services. The default value is "9093".
```
Comment (ran-huang) [inline tiup/tiup-dm-topology-reference.md:263]:
```suggestion
- `web_port`: specify the port on which Alertmanager provides web services. The default value is "9093".
```

File: tiup/tiup-dm-topology-reference.md [hunk 18]
```diff
@@ -0,0 +1,293 @@
...
+- `cluster_port`: Specify the port on which Alertmanger communicates with Alertmanagers. The default value is "9094".
```
Comment (ran-huang) [inline tiup/tiup-dm-topology-reference.md:264]:
```suggestion
- `cluster_port`: Specify the communication port between one Alertmanger and other Alertmanager. The default value is "9094".
```

**Review by ran-huang — COMMENTED** [review]
Rest LGTM
A global comment: please limit the use of future tenses.

**Pull request merged**